src/mprk.jl

```diff
@@ -379,10 +320,10 @@ function perform_step!(integrator, cache::MPECache, repeat_step = false)
     fill!(P, zero(eltype(P)))
 
     f.p(P, uprev, p, t) # evaluate production terms
-    sum_destruction_terms!(D, P) # store destruction terms in D
     integrator.stats.nf += 1
 
-    build_mprk_matrix!(P, 1, P, D, uprev, dt)
+    build_mprk_matrix!(P, P, uprev, dt)
+
```

Comment:
Do we need `D` (unpacked from the `cache` above) at all after this change? If not, should we just get rid of it?